src/resources/auth/auth.controller.ts

```diff
@@ -26,17 +26,45 @@ const logger = createLogger(module)
  * @param res
  * @param next
  */
-export const signup: RequestHandler = (req, res, next) => {
+export const signup: RequestHandler = async (req, res, next) => {
 	logger.debug('Sign up with: %o', req.body)
 
-	const user = req.body
+	const newUser = req.body
 
-	createUser(user)
-		.then(user => {
-			const token = newToken(user)
-			return res.json(successResponse({token}, true))
-		})
-		.catch(err => next(apiError.badRequest(err.message)))
+	try {
+		const user = await createUser(newUser)
+
+		// Generate reset token
+		const {resetToken, resetTokenExp} = generateResetToken()
+		user.resetToken = resetToken
+		user.resetTokenExp = resetTokenExp
+
+		// Save user to the database
+		await saveUser(user)
+
+		logger.debug('Save user: ', user.email)
+
+		// Send an email to user, containing the activation link
+		const activeUrl = `${req.headers.host}/auth/active/${user.resetToken}`
+
+		const message: Message = {
+			from: config.mailSender,
+			to: user.email,
+			subject: 'Activate your account',
+			text: `To active your account, please click the following link: \n \n
+				${activeUrl}
+			`,
+		}
+
+		await sendEmail(message)
+
+		logger.debug('Send reset password link to email: ', user.email)
+
+		const token = newToken(user)
+		return res.json({token})
```

Comment:
It is fine here but should we use `successResponse({token})` here for consistency?
Also, I think we only need to send a message response to tell user to check their email, like what we did in `resetPassword`